### PR TITLE
[DO NOT MERGE] multiplayer version 4: part 3: check Console ID conflicts

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
**DO NOT MERGE OR PUT THIS ON CANARY. THIS REQUIRES A VERSION BUMP.**
When this is reviewed, I (or the maintainers) can push it to #4377.
To allow for easier compare, visit https://github.com/zhaowenlan1779/citra/compare/multiplayer-v4/main...zhaowenlan1779:multiplayer-v4/feature/console-id-protection

---

This should be pretty straightforward.

Some users are having identical Console IDs in the rooms. This may confuse games and cause issues with multiplayer. This PR checks and prevents the issue.

@chris062689 thinks that Console ID can be sensitive data sometimes, so currently I'm sending a SHA256 hash of it. (actually I'm not sure if this is necessary)

Screenshot:
image

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4390)

### CODE REVIEW

**Comment by chris062689** [comment]
Minor nits. 

"someone else's" shouldn't this be "someone else"? 

Also, can be there be two line breaks between the error message and the action they should take?

Your Console ID conflicted with someone else in the room. 

Please go to...

**Comment by zhaowenlan1779** [comment]
> "someone else's" shouldn't this be "someone else"?

Hmm.. I was thinking it should be `sth conflict with sth`...

File: src/core/hle/service/cfg/cfg.cpp [hunk 1]
```diff
@@ -734,4 +734,21 @@ void InstallInterfaces(Core::System& system) {
     std::make_shared<CFG_NOR>()->InstallAsService(service_manager);
 }
 
+std::string GetConsoleIdHash(Core::System& system) {
+    std::array<u8, sizeof(u64)> buffer;
+    u64 console_id{};
```
Comment (wwylele) [inline src/core/hle/service/cfg/cfg.cpp]: nit: as you access the underlying storage of this value, it is better to specify it's endianness (`_le` or `_be`).

**Comment by zhaowenlan1779** [comment]
Pushed to #4377.

**Pull request closed without merging**